File: EtwEvents.AgentManager/src/js/eventSinkProfile.js
class EventSinkProfile {
  constructor(name, sinkType, version, options, credentials) {
    this.name = name;
    this.sinkType = sinkType;
    this.version = version;
    this.options = options || {};
    this.credentials = credentials || {};
  }
}

export default EventSinkProfile;

File: EtwEvents.Client.Shared/EventSinkProfile.cs
using System.Collections.Generic;

namespace KdSoft.EtwEvents.Client.Shared
{
    public class EventSinkProfile
    {
        public string SinkType { get; set; } = nameof(DummySink);
        public string Name { get; set; } = "Dummy";
        public string Version { get; set; } = "1.0";
        public Dictionary<string,object> Options { get; set; } = new Dictionary<string, object>();
        public Dictionary<string, object> Credentials { get; set; } = new Dictionary<string, object>();
    }
}
